components/calendar/src/calendar.rs

```diff
@@ -12,17 +12,27 @@ use crate::{Date, DateDuration, DurationUnit, Iso};
 /// Individual [`Calendar`] implementations may have inherent utility methods
 /// allowing for direct construction, etc.
 pub trait Calendar {
+    /// The internal type used to represent dates
     type DateInner: PartialEq + Eq + Clone;
     /// Construct the date from an ISO date
     fn date_from_iso(&self, iso: Date<Iso>) -> Self::DateInner;
+    /// Obtain an ISO date from this date
     fn date_to_iso(&self, date: &Self::DateInner) -> Date<Iso>;
     // fn validate_date(&self, e: Era, y: Year, m: MonthCode, d: Day) -> bool;
     // // similar validators for YearMonth, etc
 
     // fn is_leap<A: AsCalendar<Calendar = Self>>(&self, date: &Date<A>) -> bool;
+    /// Count the number of months in a given year, specified by providing a date
+    /// from that year
```

Comment:
Thought: I'd love if we could give examples for these (although that would require implementing something like the Julian calendar to make it interesting)

Reply:
Note that the Calendar trait isn't really the public API, the Date type is. But I'll add examples there later.